deploy/ForeignControllerInit.sol

```diff
@@ -92,55 +115,50 @@ library ForeignControllerInit {
-        require(address(newController.cctp()) == checkAddresses.cctp, "ForeignControllerInit/incorrect-cctp");
+            require(address(newController.psm())  == params[i].checkAddresses.psm,  "ForeignControllerInit/incorrect-psm");
+            require(address(newController.usdc()) == params[i].checkAddresses.usdc, "ForeignControllerInit/incorrect-usdc");
+            require(address(newController.cctp()) == params[i].checkAddresses.cctp, "ForeignControllerInit/incorrect-cctp");
 
-        require(configAddresses.oldController != address(newController), "ForeignControllerInit/old-controller-is-new-controller");
+            require(params[i].configAddresses.oldController != address(newController), "ForeignControllerInit/old-controller-is-new-controller");
 
-        // Step 2: Perform PSM sanity checks
+            // Step 2: Perform PSM sanity checks
 
-        IPSM3Like psm = IPSM3Like(checkAddresses.psm);
+            IPSM3Like psm = IPSM3Like(params[i].checkAddresses.psm);
 
-        require(psm.totalAssets() >= 1e18, "ForeignControllerInit/psm-totalAssets-not-seeded");
-        require(psm.totalShares() >= 1e18, "ForeignControllerInit/psm-totalShares-not-seeded");
+            require(psm.totalAssets() >= 1e18, "ForeignControllerInit/psm-totalAssets-not-seeded");
+            require(psm.totalShares() >= 1e18, "ForeignControllerInit/psm-totalShares-not-seeded");
 
-        require(psm.usdc()  == checkAddresses.usdc,  "ForeignControllerInit/psm-incorrect-usdc");
-        require(psm.usds()  == checkAddresses.usds,  "ForeignControllerInit/psm-incorrect-usds");
-        require(psm.susds() == checkAddresses.susds, "ForeignControllerInit/psm-incorrect-susds");
+            require(psm.usdc()  == params[i].checkAddresses.usdc,  "ForeignControllerInit/psm-incorrect-usdc");
+            require(psm.usds()  == params[i].checkAddresses.usds,  "ForeignControllerInit/psm-incorrect-usds");
+            require(psm.susds() == params[i].checkAddresses.susds, "ForeignControllerInit/psm-incorrect-susds");
 
-        // Step 3: Configure ACL permissions controller, almProxy, and rateLimits
+            // Step 3: Configure ACL permissions controller, almProxy, and rateLimits
 
-        IALMProxy   almProxy   = IALMProxy(controllerInst.almProxy);
-        IRateLimits rateLimits = IRateLimits(controllerInst.rateLimits);
+            IALMProxy   almProxy   = IALMProxy(params[i].controllerInst.almProxy);
+            IRateLimits rateLimits = IRateLimits(params[i].controllerInst.rateLimits);
 
-        newController.grantRole(newController.FREEZER(), configAddresses.freezer);
-        newController.grantRole(newController.RELAYER(), configAddresses.relayer);
+            newController.grantRole(newController.FREEZER(), params[i].configAddresses.freezer);
+            newController.grantRole(newController.RELAYER(), params[i].configAddresses.relayer);
```

Comment:
This needs to be refactored. This is the only line that actually matters, should be able to pass in an array of addresses for `params.configAddresses` so that multiple `relayer`s can be given the `RELAYER` role. Can just convert this one line to a for loop and iterate through them all.

Reply:
Same thing for mainnet controller

Reply:
here in that the `params[i].configAddresses` also needs to be changed and it is a struct therefore doesn't seem well for a for loop
![Screenshot 2025-04-29 at 1 50 36 PM](https://github.com/user-attachments/assets/4abad8a8-0846-4c7e-82fc-b23bf60dc988)